goal_src/jak2/engine/target/gun/gun-h.gc

```diff
@@ -5,5 +5,322 @@
+   (fire-range             meters                :offset 296)
+   (laser-active?          symbol                :offset 300)
+   (laser-point            vector                :offset 304)
+   (laser-dir              vector                :offset 320)
+   (laser-hit-point        vector                :offset 352)
+   (track?                 uint16                :offset 368)
+   (track-tilt             degrees               :offset 372)
+   (track-turn             degrees               :offset 376)
+   (track-find-range       meters                :offset 380)
+   (track-turnv-range      meters                :offset 384)
+   (track-tilt-range       meters                :offset 388)
+   (track-turn-range       meters                :offset 392)
+   (track-tilt-max         degrees               :offset 396)
+   (track-turn-max         degrees               :offset 400)
+   (track-angle-mult       float                 :offset 404)
+   (track-beam-size        float                 :offset 408)
+   (track-auto-fire        symbol                :offset 412)
+   (track-require          uint32                :offset 416)
+   (track-target-hold-time time-frame            :offset 424)
+   (track-start-time       time-frame            :offset 432)
+   (track-press-start-time time-frame            :offset 440)
+   (track-target           focus         2       :offset 448)
+   (track-trans            vector                :offset 480)
+   (track-dir              vector                :offset 496)
+   (turn-fast-hold-time    time-frame            :offset 512)
+   (blue-whine-sound-id    uint32                :offset 520)
+   (blue-whine-volume      float                 :offset 524)
+   (top-anim-twist         vector                :offset 528)
+   (top-anim-twist-targ    vector                :offset 544)
+   (top-anim-look-at       vector                :offset 560)
+   (top-anim-twist-reset   uint64                :offset 576)
+   (top-anim-gun-height    meters                :offset 584)
+   (top-anim-blue-cycle    float                 :offset 588)
+   (top-anim-low-high      float                 :offset 592)
+   (top-anim-extra-twistv  degrees               :offset 596)
+   (top-anim-tilt-up       degrees               :offset 600)
+   (attack-combo           combo-tracker         :offset 608)
+   (combo-window-start     time-frame            :offset 656)
+   (combo-window-state     uint32                :offset 664)
+   (combo-fire-delay       uint32                :offset 668)
+   (charge-ammo            float                 :offset 672)
+   (charge-start-time      time-frame            :offset 680)
+   (charge-inc-time        time-frame            :offset 688)
+   (charge-active?         uint64                :offset 696)
+   )
+  :method-count-assert 10
+  :size-assert         #x2c0
+  :flag-assert         #xa000002c0
+  (:methods
+    (gun-info-method-9 () none 9)
+    )
+  )
+
+(defun want-to-gun? ((arg0 target) (arg1 int))
```

Comment:
I think this function would come out nicer if you set the type as
```
(define-extern want-to-gun? (function target symbol symbol :behavior process))
```
Making something a `behavior` lets it access the current process (`pp`) through a special `self` variable, and generally looks nicer. The usual guideline is to make something a `behavior` if it's not actually changing the value of `pp` itself, and just reading and writing fields of the process.

The `arg1` type as a `symbol` will clean up the `the-as symbol` cast.
If you search for uses of this function, they are all like
```
    lw t9, want-to-gun?(s7)   ;; [190] (set! t9 want-to-gun?)
    or a0, gp, r0             ;; [191] (set! a0 gp)
    daddiu a1, s7, #t         ;; [192] (set! a1 #t)
    jalr ra, t9               ;; [193] (call!)
    ```
and 
```
    lw t9, want-to-gun?(s7)
    or a0, s6, r0
    or a1, s7, r0
    jalr ra, t9
    sll v0, ra, 0
```
so it looks like its always `#t` or `#f`.

Reply:
Ah, I didn't think of using `process` as the type for the behavior, I figured it was a behavior for `target` at first, but that didn't work very well, this makes sense though. Output was cleaned up immensely after this and changing the second arg to `symbol` (I had it as int as first as a placeholder, but forgot to look for uses of the function so I didn't get the proper arg type).